core/variables.js

```diff
@@ -285,25 +285,33 @@ Blockly.Variables.renameVariable = function(workspace, variable,
  * Prompt the user for a new variable name.
  * @param {string} promptText The string of the prompt.
  * @param {string} defaultText The default value to show in the prompt's field.
+ * @param {string} opt_type Optional type of variable, like 'string' or 'list'.
  * @param {function(?string)} callback A callback. It will be passed the new
  *     variable name, or null if the user picked something illegal.
  */
-Blockly.Variables.promptName = function(promptText, defaultText, callback) {
+Blockly.Variables.promptName = function(promptText, defaultText, opt_type, callback) {
```

Comment:
If the new parameter is optional it has to be the last parameter--otherwise if it's left out it gets replaced by the next param, which is (almost) never what you want.

Reply:
done.